File: app/src/main/java/it/amonshore/comikkua/ui/releases/AReleaseViewModelItemViewHolder.java
package it.amonshore.comikkua.ui.releases;

import android.view.View;

import com.bumptech.glide.RequestManager;

import androidx.annotation.NonNull;
import it.amonshore.comikkua.data.release.IReleaseViewModelItem;
import it.amonshore.comikkua.ui.IViewHolderWithDetails;

public abstract class AReleaseViewModelItemViewHolder extends IViewHolderWithDetails<Long> {

    public AReleaseViewModelItemViewHolder(View itemView) {
        super(itemView);
    }

    public abstract void bind(@NonNull IReleaseViewModelItem item, boolean selected, RequestManager requestManager);

    public abstract void clear();
}

File: app/src/main/java/it/amonshore/comikkua/ui/releases/adapter/AReleaseViewModelItemViewHolder.kt
package it.amonshore.comikkua.ui.releases.adapter

import android.view.View
import com.bumptech.glide.RequestManager
import it.amonshore.comikkua.data.release.IReleaseViewModelItem
import it.amonshore.comikkua.ui.IViewHolderWithDetails

abstract class AReleaseViewModelItemViewHolder(itemView: View) :
    IViewHolderWithDetails<Long>(itemView) {

    abstract fun bind(
        item: IReleaseViewModelItem,
        selected: Boolean = false,
        glide: RequestManager? = null,
        onReleaseClick: OnReleaseClick? = null,
        onReleaseMenuClick: OnReleaseMenuClick? = null
    )

    abstract fun clear()
}